src/BasicBehaveEngine/nodes/math/bitwise/Xor.ts

```diff
@@ -0,0 +1,36 @@
+import {BehaveEngineNode, IBehaviourNodeProps} from "../../../BehaveEngineNode";
+
+export class Xor extends BehaveEngineNode {
+    REQUIRED_VALUES = [{id:"a"}, {id:"b"}]
+
+    constructor(props: IBehaviourNodeProps) {
+        super(props);
+        this.name = "XorNode";
+        this.validateValues(this.values);
+    }
+
+    override processNode(flowSocket?: string) {
+        const {a, b} = this.evaluateAllValues(this.REQUIRED_VALUES.map(val => val.id));
+        this.graphEngine.processNodeStarted(this);
+        const typeIndexA = this.values['a'].type!
+        const typeA: string = this.getType(typeIndexA);
+        const typeIndexB = this.values['b'].type!
+        const typeB: string = this.getType(typeIndexB);
+        if (typeA !== typeB) {
+            throw Error("input types not equivalent")
+        }
+
+        let val: any;
+        switch (typeA) {
+            case "bool":
+                val = (JSON.parse(a) && !JSON.parse(b)) || (!JSON.parse(a) && JSON.parse(b));
+                break;
```

Comment:
Boolean (conditional) XOR is not in the spec because it's rarely present in programming languages.

If we need to have it, the implementation should be just `JSON.parse(a) != JSON.parse(b)`.

Reply:
I would like to have it for completeness of full expressiveness of any business logic, (i.e. only open my door if I have clicked block 1 or block 2 but not both) I'll update the implementation in the next push

Reply:
OK, the spec has been updated.